charts/spire/charts/spire-agent/templates/daemonset.yaml

```diff
@@ -170,6 +177,15 @@ spec:
         - name: spire-config
           configMap:
             name: {{ include "spire-agent.fullname" . }}
+        {{- if .Values.sockets.admin.mountOnHost }}
```

Comment:
The client for the spire-agent UDS is the executable `spire-agent`.  I don't think we should expose the socket using `hostPath` as the only deployed client will exist within the spire-agent Pod.

Reply:
Addressed above

Reply:
I'd rather see this added with Cilium's PR, so we can review it all in one context that has something to do with Cilium.  When we review Cilium supporting changes, because this will already be merged into the baseline, we likely will miss any issues that tie back to this change.